src/util/Serializer/Serializer.h

```diff
@@ -242,8 +244,9 @@ CPP_requires(
 
 template <typename T>
 CPP_concept TriviallySerializable =
-    CPP_requires_ref(triviallySerializableRequires, T) &&
-    std::is_trivially_copyable_v<std::decay_t<T>>;
+    (CPP_requires_ref(triviallySerializableRequires, T) &&
+     std::is_trivially_copyable_v<std::decay_t<T>>) ||
+    std::is_enum_v<std::decay_t<T>>;
```

Comment:
Enums are trivially copyable right? There's just something wrong with the templates I guess?

Reply:
I have moved the `is_enum_v` to a better place.